cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocUtils.java

```diff
@@ -274,9 +284,17 @@ public static String getSystemZone(ConnectionProvider connectionProvider) {
       String zone = getMetadata(connectionProvider, "instance/zone");
       // will be something like projects/<project-number>/zones/us-east1-b
       return zone.substring(zone.lastIndexOf('/') + 1);
-    } catch (IOException e) {
-      throw new IllegalArgumentException("Unable to get the zone from the environment. "
-          + "Please explicitly set the zone.", e);
+    } catch (Exception e) {
+      String errorReason = "Unable to get the zone from the environment. "
+          + "Please explicitly set the zone.";
+      throw new DataprocRuntimeException.Builder()
+          .withCause(e)
+          .withErrorCategory(DataprocRuntimeException.ERROR_CATEGORY_CONFIGURATION)
```

Comment:
nit: This method, BTW, is called from DataprocConf.create that is pretty expensive and is done pretty often. Also it's done in various situations. In most cases it would be original provisioning though and it would take too much refactoring to provide proper phase (provisioning / deprovisioning / other) information, so I am fine leaving it as is.
If you wish to make it better, I'd probably remove the connection provider parameter as it's always same and cache the value after we got it first time. It won't change for given VM.Ditto for system project id

Reply:
left it as is for now.